Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,123 @@ public class UserCommunicateService : IUserCommunicateService
 
-        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService)
+        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService, IValidator<ForecastWeatherRequestDTO> validator)
         {
             _logger = logger;
             _weatherServiсe = weatherService;
+            _validator = validator;
+            _invoker = new Invoker();
         }
 
-        public async Task<bool> CommunicateAsync()
+        public async Task StartUpApplicationAsync()
         {
-            Console.WriteLine("Please, enter city name:");
+            Console.WriteLine("Select menu item:");
+            Console.WriteLine("0 - Get currently weather");
+            Console.WriteLine("1 - Get weather for a period of time");
+            Console.WriteLine("2 - Exit");
+
+            bool parseResult = int.TryParse(Console.ReadLine(), out var pointMenu);
 
-            var cityName = Console.ReadLine();
+            if (!parseResult)
+            {
+                Console.WriteLine(Constants.Errors.IncorrectValue);
+                _logger.LogError($"User entered incorrect value.");
+                return;
+            }
 
-            if (cityName.ToUpper() == Constants.Commands.Exit)
+            if (pointMenu < 0 || pointMenu > 2)
             {
-                return false;
+                Console.WriteLine(Constants.Errors.IncorrectValue);
+                _logger.LogError($"User entered value out of range.");
+                return;
             }
 
-            if (string.IsNullOrEmpty(cityName))
+            switch (pointMenu)
             {
-                Console.WriteLine(Constants.Validation.EmptyCityName);
-                _logger.LogInformation("The user entered an empty city name. Try again");
-                return true;
+                case 0:
+                    _invoker.SetCommand(new CurrentlyWeatherCommand(this));
+                    break;
+                case 1:
+                    _invoker.SetCommand(new ForecastWeatherCommand(this));
+                    break;
+                case 2:
+                    _invoker.SetCommand(new ExitCommand());
+                    break;
             }
 
-            var weatherRepresentation = (string)default;
             try
             {
-                weatherRepresentation = (await _weatherServiсe.GetByCityNameAsync(cityName)).GetStringRepresentation();
+                await _invoker.RunAsync();
             }
             catch (HttpRequestException ex)
             {
                 if (ex.StatusCode == HttpStatusCode.NotFound)
                 {
                     Console.WriteLine(Constants.Errors.BadCityName);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
-                    return true;
+                    _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. User entered incorrect city name.");
```

Comment:
I suggested to use StatusCodes class, where is it

Reply:
+